components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java

```diff
@@ -1943,31 +1945,29 @@ private OMElement queryBetweenTwoDaysForFaulty(String columnFamily, String fromD
         Connection connection = null;
         Statement statement = null;
         ResultSet rs = null;
+        List<APIResponseFaultCount> faultusage = new ArrayList<APIResponseFaultCount>();
+
         try {
             connection = dataSource.getConnection();
             statement = connection.createStatement();
             String query;
 
             query = "SELECT api,version,apiPublisher,context,SUM(total_fault_count) as total_fault_count FROM  "
-                    + columnFamily + " WHERE " + APIUsageStatisticsClientConstants.TIME + " BETWEEN " +
+                    + tableName + " WHERE " + APIUsageStatisticsClientConstants.TIME + " BETWEEN " +
                     "\'" + fromDate + "\' AND \'" + toDate + "\'" + " GROUP BY api,version,apiPublisher,context";
 
             rs = statement.executeQuery(query);
-            StringBuilder returnStringBuilder = new StringBuilder("<omElement><rows>");
-            int columnCount = rs.getMetaData().getColumnCount();
+            APIResponseFaultCount apiResponseFaultCount;
+
             while (rs.next()) {
-                returnStringBuilder.append("<row>");
-                for (int i = 1; i <= columnCount; i++) {
-                    String columnName = rs.getMetaData().getColumnName(i);
-                    String columnValue = rs.getString(columnName);
-                    returnStringBuilder.append("<" + columnName.toLowerCase() + ">" + columnValue +
-                            "</" + columnName.toLowerCase() + ">");
-                }
-                returnStringBuilder.append("</row>");
+                String apiName = rs.getString("api");
+                String version = rs.getString("version");
```

Comment:
Shall we use constants and use them because we can see that "api", "version", "context" etc.. is used many times in this code.